src/main/java/common/tileentities/GTMTE_LapotronicSuperCapacitor.java

```diff
@@ -233,14 +233,16 @@ public boolean checkMachine(IGregTechTileEntity thisController, ItemStack guiSlo
 					final int meta = thisController.getMetaIDOffset(offset.x(), offset.y(), offset.z());
 					if(thisController.getBlockOffset(offset.x(), offset.y(), offset.z()) == LSC_PART && (meta > 0)) {
 						// Add capacity
-						if(meta <= 4){
-							final long c = (long) (100000000L * Math.pow(10, meta - 1));
-							tempCapacity = tempCapacity.add(BigInteger.valueOf(c));
-							capacity = capacity.add(BigInteger.valueOf(c));
-						} else if(meta <= 5){
-							tempCapacity = tempCapacity.add(BigInteger.valueOf((long) (100000000L * Math.pow(10, 3))));
-							capacity = capacity.add(MAX_LONG);
-						}
+						switch(meta - 1) {
```

Comment:
Why `- 1`?

Reply:
Because of the ```final long c = (long) (100000000L * Math.pow(10, meta - 1));``` line. But I guess that's wrong, and I didn't notice it because the capacity is defined again correctly lower down.